table/block_fetcher.cc

```diff
@@ -178,14 +189,21 @@ inline void BlockFetcher::GetBlockContents() {
     // page can be either uncompressed or compressed, the buffer either stack
     // or heap provided. Refer to https://github.com/facebook/rocksdb/pull/4096
     if (got_from_prefetch_buffer_ || used_buf_ == &stack_buf_[0]) {
-      CopyBufferToHeap();
+      CopyBufferToHeapBuf();
     } else if (used_buf_ == compressed_buf_.get()) {
       if (compression_type_ == kNoCompression &&
           memory_allocator_ != memory_allocator_compressed_) {
-        CopyBufferToHeap();
+        CopyBufferToHeapBuf();
       } else {
         heap_buf_ = std::move(compressed_buf_);
       }
+    } else if (direct_io_buf_.get() != nullptr) {
+      if (compression_type_ == kNoCompression) {
+        CopyBufferToHeapBuf();
+      } else {
+        CopyBufferToCompressedBuf();
```

Comment:
Previously, ```memory_allocator_``` was used if ```do_uncompress_``` was true, and ```memory_allocator_compressed_``` was used if ```do_uncompress_``` was false. Should the same check be done here? My guess is ```memory_allocator_compressed_``` is used if the block needs to be inserted in the compressed block cache?

Reply:
If the block needs to be uncompressed, it will be uncompressed in https://github.com/facebook/rocksdb/pull/6689/files#diff-e43ed8357174b6496b131fbf68e138b4R295.

Entering this code path means the block does not need to be uncompressed, so if compression type is NoCompression, then insert to heap buf, otherwise insert to compressed buf.

Reply:
Good point. Thanks for the clarification.